modules/bigquery-dataset/main.tf

```diff
@@ -267,3 +268,46 @@ resource "google_bigquery_table" "views" {
     use_legacy_sql = each.value.use_legacy_sql
   }
 }
+
+resource "google_bigquery_table" "materialized_view" {
+  depends_on          = [google_bigquery_table.default]
+  for_each            = var.materialized_views
+  project             = var.project_id
+  dataset_id          = google_bigquery_dataset.default.dataset_id
+  table_id            = each.key
```

Comment:
can you use an implicit reference here to the table resource, and remove the explicit dependency above?

Reply:
Shoul it be done also for the `views?` I followed the same approach.
Here the code: https://github.com/devuonocar/cloud-foundation-fabric/blob/97b8af8c17a836fc1e296454d6261893698293bf/modules/bigquery-dataset/main.tf#L260C8-L260C8

Reply:
Ah yes, we always prefer implicit dependencies